### PR TITLE
Fixed implementation of Alternative for EitherT

### LINKED ISSUE #46: Fix Alternative for EitherT
It runs monad stack more than once

### PR DESCRIPTION
**Pull request by sergei-shabanau**
Closes #46 

@soujiro32167 It was my 🐛😬, what else could it be 😼

`Alternative.scala` has the fix.

@regis-leray we would need to redeploy the lib.

### CODE REVIEW

File: src/main/scala/scalaz/parsers/Parsing.scala [hunk 1]
```diff
@@ -71,6 +71,12 @@ sealed trait Parsing[F[_], G[_], E] {
         b => if (p(b)) G.pure(b) else G.raiseError(e)
       )
 
+    def ensure[A](p: A => Boolean)(implicit E: Monoid[E]): Equiv[A, A] =
```
Comment (ryanonsrc) [inline src/main/scala/scalaz/parsers/Parsing.scala]: 👍 for introducing a Monoid

**Comment by ryanonsrc** [comment]
looks good.  After we get at least one more approval either myself or @sergei-shabanau can feel free to merge.

**Comment by soujiro32167** [comment]
Is this the 2^🐤bug?

File: src/main/scala/scalaz/parsers/tc/Alternative.scala [hunk 2]
```diff
@@ -27,9 +26,6 @@ object Alternative {
   implicit def eitherTAlternative[F[_]: Monad, L]: Alternative[EitherT[L, F, ?]] =
     new Alternative[EitherT[L, F, ?]] {
       override def or[A](f1: EitherT[L, F, A], f2: => EitherT[L, F, A]): EitherT[L, F, A] =
-        EitherT(f1.run.flatMap {
-          case -\/(_) => f2.run
-          case \/-(_) => f1.run
-        })
+        f1.orElse(f2)
```
Comment (soujiro32167) [inline src/main/scala/scalaz/parsers/tc/Alternative.scala]: Is this where the singularity happened?

Comment (sergei-shabanau) [reply src/main/scala/scalaz/parsers/tc/Alternative.scala]: Yes. `run` was responsible for running 😜

**Pull request merged**